From 0c088305fbdcfc22afa8383204314548807671ee Mon Sep 17 00:00:00 2001
From: Stanislav Laznicka <slaznick@redhat.com>
Date: Thu, 19 May 2022 13:49:49 +0200
Subject: [PATCH] UPSTREAM: <carry>: e2e-framework: don't autosync PodSecurity
 labels

In the tests, we oftentimes create pods directly by the administrative
user and so their SCC-related privileges are being used to create the
pods. The PSa label syncher however works by introspecting SAs in each
namespace, and since the SAs in the direct pod creation use-cases don't
have the SCC-related privileges, the labelsyncer evaluates these
namespaces as "restricted" because only the "restricted-v2" SCC is ever
assigned in the namespaces. This breaks tests where pods are created
directly.

OpenShift-Rebase-Source: 4b7ae5621c4
---
 test/e2e/framework/framework.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/e2e/framework/framework.go b/test/e2e/framework/framework.go
index e29e0edf58e..99cca71f70e 100644
--- a/test/e2e/framework/framework.go
+++ b/test/e2e/framework/framework.go
@@ -460,6 +460,9 @@ func (f *Framework) CreateNamespace(ctx context.Context, baseName string, labels
 	labels[admissionapi.EnforceLevelLabel] = firstNonEmptyPSaLevelOrRestricted(f.NamespacePodSecurityEnforceLevel, f.NamespacePodSecurityLevel)
 	labels[admissionapi.WarnLevelLabel] = firstNonEmptyPSaLevelOrRestricted(f.NamespacePodSecurityWarnLevel, f.NamespacePodSecurityLevel)
 	labels[admissionapi.AuditLevelLabel] = firstNonEmptyPSaLevelOrRestricted(f.NamespacePodSecurityAuditLevel, f.NamespacePodSecurityLevel)
+	// turn off the OpenShift label syncer so that it does not attempt to sync
+	// the PodSecurity admission labels
+	labels["security.openshift.io/scc.podSecurityLabelSync"] = "false"
 
 	ns, err := createTestingNS(ctx, baseName, f.ClientSet, labels)
 	// check ns instead of err to see if it's nil as we may
-- 
2.41.0

